stix_shifter/stix_translation/src/modules/msatp/query_constructor.py

```diff
@@ -0,0 +1,406 @@
+            if index_of_field < len(mapped_fields_array) - 1:
+                comparison_string += " or "
+        return comparison_string
+
+    @staticmethod
+    def clean_format_string(format_string):
+        """
+        Formats and replaces carriage return(\r), newline character(\n), spaces > 2, tab with 1 space
+        :param format_string: str
+        :return: str
+        """
+        return re.sub(r'\r|\n|\s{2,}|\t', ' ', format_string)
+
+    def get_lookup_table_of_obs_exp(self, expression, lookup_table, objects_dict):
+        """
+        Function to parse observation expression and return the object(i.e FileCreationEvents
+        , ProcessCreationEvents, NetworkCommunicationEvents, RegistryEvents) involved
+        :param expression: expression object, ANTLR parsed expression object
+        :param lookup_table: list, empty list
+        :param objects_dict: dict, dictionary of objects as keys and respective fields as values
+        :return: None
+        """
+
+        if hasattr(expression, 'object_path'):
+            stix_object = expression.object_path.split(':')[0]
+            value = expression.value.values if hasattr(expression.value, 'values') else expression.value
+            value_type = self._check_value_type(value, expression)
+            if 'mac' in value_type:
+                self._is_mac = True
+            lookup_table.append(objects_dict.get(stix_object))
+            final_lookup = list(set(lookup_table))
+            if self._is_mac:
+                self.lookup_table_object = 'NetworkCommunicationEvents'
+            else:
+                if len(final_lookup) == 1:
+                    self.lookup_table_object = final_lookup[0]
+                elif 'NetworkCommunicationEvents' in final_lookup:
+                    self.lookup_table_object = 'NetworkCommunicationEvents'
+                elif 'RegistryEvents' in final_lookup:
+                    self.lookup_table_object = 'RegistryEvents'
+                elif 'ProcessCreationEvents' in final_lookup:
+                    self.lookup_table_object = 'ProcessCreationEvents'
+
+        else:
+            self.get_lookup_table_of_obs_exp(expression.expr1, lookup_table, objects_dict)
+            self.get_lookup_table_of_obs_exp(expression.expr2, lookup_table, objects_dict)
+
+    def _lookup_comparison_operator(self, expression_operator):
+        """
+        Comparison operator lookup with error handling for un-supported operators
+        :param expression_operator: str
+        :return: str
+        """
+        if expression_operator not in self.comparator_lookup:
+            raise NotImplementedError(
+                "Comparison operator {} unsupported for MSATP adapter".format(expression_operator.name))
+        return self.comparator_lookup.get(expression_operator)
+
+    def _parse_expression(self, expression, qualifier=None):
+        """
+        Complete formation of relevance query from ANTLR expression object
```

Comment:
I see the word `relevance` is used in many places of query_constructor.py file. I think it is copied from BigFix(relevance is the query language of BigFix). you need to replace that with whatever is appropriate for msatp. Usually we use the query name of the api or something that refers the datasource/api.

Reply:
Updated relevance in query_constructor.py